### PR TITLE
Fix sphinx-docs warning

### PR DESCRIPTION
**Pull request by mojotx**
When building from source, there is a warning:

     ../doc_src/cmds/string-match.rst:13: WARNING: Inline emphasis start-string without end-string.

One fix appears to be putting a space after the emphasized 'n' character, e.g., `*n* th` instead of `*n*th`.

## Description

I'm not an expert with ReStructuredText, but this seems to be a reasonable way to silence the warning. 

Fixes issue #7519 

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.  _(N/A)_
- [ ] Tests have been added for regressions fixed. _(N/A)_
- [ ] User-visible changes noted in CHANGELOG.rst _(N/A)_

### CODE REVIEW

File: doc_src/cmds/string-match.rst [hunk 1]
```diff
@@ -27,7 +27,7 @@ If ``--index`` or ``-n`` is given, each match is reported as a 1-based start pos
...
 
-When matching via regular expressions, it is possible to directly import matches as fish variables by means of named capture groups using the PCRE2 syntax. This behavior is automatic and occurs in addition to any other match reporting. The default behavior with ``--regex`` results in the initialization of fish variables in the default scope containing the matched text corresponding to each named capture group. A named capture group matching a zero-length string will be initialized as a fish variable containing a likewise empty string (i.e. the equivalent of ``""``), but a named capture group that did not match will result in an empty (null) fish variable. When ``--regex`` is used in conjunction with ``--all``, this behavior changes slightly: for each of the *n* matching sequences, the *n*th index of each named variable will contain the contents of the corresponding named capture group for the *n*th match, but will contain an empty string if the group was empty or if the group did not match.
```
Comment (charego) [inline doc_src/cmds/string-match.rst:30]: The literal space looks a little funky: `*n* th`

nth-spaces

Escaping looks slightly better to me: `*n*\ th`

nth-escapes

A superscript looks cool but messes up the line height:
```
*n*:superscript:`th`
```

nth-superscripts

**Comment by zanchey** [comment]
What about rewording? "When ``--regex`` is used with ``--all``, this behavior changes. Each named variable will contain a list of matches, with the first match contained in the first element, the second match in the second, and so on. If the group was empty or did not match, the corresponding element will be an empty string."

**Comment by mqudsi** [comment]
I'm good with @zanchey's suggestion, but would still like to know how the original could be done. As I said in the issue, `pandoc` renders it correctly (or at least, as intended).

**Comment by charego** [comment]
@mqudsi https://github.com/fish-shell/fish-shell/pull/7520#discussion_r533095541 tl;dr: `*n*\ th`

**Comment by mojotx** [comment]
I just pushed a change (3c11ea2a5a62720527e44e44c5addbe06a9e36fd), escaping the space.

**Comment by charego** [comment]
I thought we agreed to reword it? You presented a nail and I had a hammer, but @zanchey's suggestion is better.

**Comment by mojotx** [comment]
@charego: I replaced the text with @zanchey's suggestion, with commit ce32c6dae19f0e05782cad11d3809cadd06e5163

**Comment by ridiculousfish** [comment]
Merged as eb22a9c4db1a2b7ffda030650ca00a393829b03b, thank you!

**Pull request closed without merging**